service-jenkins-cleanup/service-jenkins-cleanup.sh

```diff
@@ -0,0 +1,221 @@
+        -o)
+          short_opt "DAYS" $@
+          shift 2
+          ;;
+        --older*)
+          long_opt "DAYS" $@
+          shift
+          ;;
+        -v|--verbose)
+          shift
+          VERBOSE=true
+          ;;
+        -d)
+          short_opt "dry_run" $@
+          shift 2
+          ;;
+        --dry_run*)
+          long_opt "dry_run" $@
+          shift
+          ;;
+        *)
+          echo "Error: Invalid command syntax"
+          help_text
+          exit 1
+          ;;
+      esac
+    done
+
+    if [[ "$dry_run" != "true" ]] && [[ "$dry_run" != "false" ]]; then
+        echo "Error: -d/--dry_run must be set to 'true' or 'false'"
+        exit 1
+    fi
+
+    # Verify creds specified
+    if [[ -z "${CLUSTER_PASSWORD}" ]]; then
+        echo "ERROR: SERVICE_CLUSTER_PASSWORD must be set."
+        exit 1
+    fi
+
+    if [[ -z "${JENKINS_API_TOKEN}" ]]; then
+        echo "ERROR: JENKINS_API_TOKEN must be set."
+        exit 1
+    fi
+}
+
+get_jenkins_pod() {
+    # Get running Jenkins pod
+    jenkins_pod=`kubectl get pod -n jenkins -l name=jenkins --field-selector status.phase=Running -o jsonpath="{.items[0].metadata.name}"`
+    if [ $? != 0 ]; then
+        echo "Error: Unable to obtain running jenkins pod"
+        exit 1
+    fi
+}
+
+
+run_cmd_on_pod() {
+    cmd_results=""
+
+    retries=4
+    delay=300
+
+    while [ $retries -gt 0 ]; do
+        cmd_results=$(kubectl exec --namespace=jenkins ${jenkins_pod} -- bash -c "${CMD}")
+        if [ $? -eq 0 ]; then
+            break # Exit loop if successful
+        else
+            echo "WARNING: Command failed, retrying in $delay seconds..."
+            sleep $delay
+            retries=$((retries - 1))
+        fi
+    done
+
+    if [ $retries -eq 0 ]; then
+        echo "ERROR: Failed to run '$CMD' on $jenkins_pod after multiple retries."
+        exit 1
+    fi
+}
+
+
+verify_syntax $@
+
+echo "dry_run    : ${dry_run}"
+echo "verbose    : ${VERBOSE}"
+echo "DAYS       : ${DAYS}"
+echo " "
+
+# Create temp KUEBCONFIG file
+mkdir /tmp/.kube
+if [ $? != 0 ]; then
+    echo "Error: Failed to create directory"
+    exit 1
+fi
+
+touch $KUBECONFIG
+if [ $? != 0 ]; then
+    echo "Error: Failed to create KUBECONFIG file - $KUBECONFIG"
+    exit 1
+fi
+
+# Logon to service cluster
+oc login $CLUSTER_URL --username $CLUSTER_USER --password $CLUSTER_PASSWORD
+if [ $? != 0 ]; then
+    echo "Error: Unable to login to SERVICE CLUSTER: ${CLUSTER_URL}"
+    exit 1
+fi
+
+get_jenkins_pod
+echo " "
+echo -e "Jenkins Pod: $jenkins_pod\n"
+
+# Get list of top level jobs directories that have not been modified in over X number of days
+CMD='find ${JENKINS_HOME}/jobs -maxdepth 1 -type d -mtime +'${DAYS}' -print'
+run_cmd_on_pod
+mapfile -d ' ' dir_list < <(printf "$cmd_results")
+
+# Go though and make sure no subdirectories have ben modified.
+for check_dir in ${dir_list[@]}; do
+    CMD="find ${check_dir} -type d -mtime -${DAYS} -print"
+    run_cmd_on_pod
```

Comment:
can you get the folders via API, instead of directly looking at pod storage? Then we can remove the creds for service cluster completely...